presentation/src/main/java/com/whyranoid/presentation/component/challenge/ChallengeGoalIndicator.kt

```diff
@@ -0,0 +1,230 @@
+package com.whyranoid.presentation.component.challenge
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.progressSemantics
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Divider
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.StrokeCap
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.drawText
+import androidx.compose.ui.text.rememberTextMeasurer
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.whyranoid.domain.model.challenge.Challenge
+import com.whyranoid.domain.model.challenge.ChallengeType
+import com.whyranoid.domain.util.EMPTY
+import com.whyranoid.domain.util.getDurationDifference
+import com.whyranoid.domain.util.getToday
+import com.whyranoid.presentation.theme.ChallengeColor.getColor
+import com.whyranoid.presentation.theme.WalkieColor
+import com.whyranoid.presentation.theme.WalkieTypography
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+@Composable
+fun ChallengeGoalIndicator(challenge: Challenge) {
+
+    val challengeColor = challenge.challengeType.getColor()
+
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clip(shape = RoundedCornerShape(15.dp))
+            .background(challengeColor.backgroundColor),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+
+        val progressBarColor = challengeColor.progressBarColor
+
+        val progress =
+            if (challenge.progress == null) 0f else requireNotNull(challenge.progress) / 100f
```

Comment:
requireNotNull을 쓰면 challenge.progress가 null이면 exception 나지 않나요? 아마 challenge.progress는 nullable한 상황이라 사용하셨을 것 같은데 다음과 같은 방식은 어떨까요?
```suggestion
 val challengeProgress = challenge.progress
 if (challengProgress == null) 0f else challenge.progress / 100f
```